File: Backend/smartbrain/src/main/java/com/smartparking/smartbrain/dto/request/Authentication/ResetPassRequest.java
package com.smartparking.smartbrain.dto.request.Authentication;

import jakarta.validation.constraints.NotNull;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ResetPassRequest {
    @NotNull(message = "User token cannot be null")
    String userToken;
    @NotNull(message = "New password cannot be null")
    String newPassword;
    @NotNull(message = "Reset token cannot be null")
    String resetToken;
}

File: Frontend/myparkingapp/lib/data/request/resetPassRequest.dart
class ResetPassRequest{
  String userToken;
  String newPassword;
  String resetToken;

  ResetPassRequest( this.newPassword,this.userToken, this.resetToken);

  Map<String, dynamic> toJson() {
    return {
      'userToken': userToken,
      'newPassword': newPassword,
      'resetToken': resetToken,
    };
  }

}